config/nginx.conf

```diff
@@ -0,0 +1,103 @@
+        add_header Strict-Transport-Security max-age=2592000;
+        rewrite ^/.*$ https://$host$request_uri? permanent;
+    }
+
+    charset utf-8;
+
+    server {
+        # server port and name
+        listen 443 ssl;
+
+        # Specifies the maximum accepted body size of a client request,
+        # as indicated by the request header Content-Length.
+        client_max_body_size 200m;
+
+        # add ssl specific settings
+        keepalive_timeout 60;
+        ssl_certificate /etc/ssl/nginx/domain.bundle.crt;
+        ssl_certificate_key /etc/ssl/nginx/domain.key;
+
+        # limit ciphers
+        ssl_ciphers 'ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:AES:CAMELLIA:DES-CBC3-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!EDH-RSA-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA';
+        ssl_protocols TLSv1.2;
+        ssl_prefer_server_ciphers on;
+        ssl_dhparam /etc/ssl/certs/dhparam.pem;
+
+        # increase proxy buffer to handle some Odoo web requests
+        proxy_buffers 16 64k;
+        proxy_buffer_size 128k;
+
+        #general proxy settings
+        # force timeouts if the backend dies
+        proxy_connect_timeout 600s;
+        proxy_send_timeout 600s;
+        proxy_read_timeout 600s;
+        proxy_next_upstream error timeout invalid_header http_500 http_502 http_503;
+
+        # set headers
+        proxy_set_header Host $host;
+        proxy_set_header X-Forwarded-Host $host;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Real-IP $remote_addr;
+
+        # Let the Odoo web service know that we’re using HTTPS, otherwise
+        # it will generate URL using http:// and not https://
+        proxy_set_header X-Forwarded-Proto $scheme;
+
+        # by default, do not forward anything
+        proxy_redirect off;
+        proxy_buffering off;
+
+        location / {
+            proxy_pass http://odoo-upstream;
+        }
+
+        location /longpolling {
+            proxy_pass http://odoo-im-upstream;
+        }
+
+        # cache some static data in memory for 60mins.
+        # under heavy load this should relieve stress on the Odoo web interface a bit.
+        location /web/static/ {
+            proxy_cache_valid 200 60m;
+            proxy_buffering on;
+            expires 864000;
+            proxy_pass http://odoo-upstream;
+        }
+        include /etc/nginx/custom_error_page.conf;
```

Comment:
Where nginx log files will be located? Some default location will be used? As I see in my local config additional logging settings are defined:
```       
        access_log /var/log/nginx/access.log;
        error_log /var/log/nginx/error.log;
```

Reply:
With docker, it works differently. It is handled by docker, using their data storage (by specific driver), by default on host. We can't store non static data and data that needs to be persisted inside container or it would be lost. 

See: https://sematext.com/blog/docker-logs-location/